### PR TITLE
fix(editor): Persist tag filter when clicking tag directly in workflows page

### PR DESCRIPTION
**Pull request by tomi**
## Summary

The filters on the Workflows page are persisted when they are changed from the filters dropdown. However they were not persisted if user clicked a tag from the workflow card. This fixes that. That way user will see the same filters when they come back to the workflows page from a workflow using back button.

Before:

https://github.com/n8n-io/n8n/assets/10324676/7718fb5f-b62d-4a91-888d-733f60099762

After:

https://github.com/n8n-io/n8n/assets/10324676/807f6048-b5dd-4bd6-9f1b-026a0cba0b15

## Related tickets and issues

https://linear.app/n8n/issue/ADO-2063/bug-clicking-on-tag-doesnt-add-tag-id-to-url

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/views/WorkflowsView.vue [hunk 1]
```diff
@@ -295,7 +301,7 @@ const WorkflowsView = defineComponent({
 	methods: {
 		onFiltersUpdated(filters: Filters) {
 			this.filters = filters;
-			this.saveFiltersOnQueryString();
+			// this.saveFiltersOnQueryString();
```
Comment (cstuncsik) [inline packages/editor-ui/src/views/WorkflowsView.vue:304]: If it is not needed just delete

File: packages/editor-ui/src/views/__tests__/WorkflowsView.test.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ describe('WorkflowsView', () => {
 	let usersStore: ReturnType<typeof useUsersStore>;
 	let projectsStore: ReturnType<typeof useProjectsStore>;
 
+	const routerReplaceMock = vi.fn();
```
Comment (cstuncsik) [inline packages/editor-ui/src/views/__tests__/WorkflowsView.test.ts:24]: Can you move this test file next to the source? (as we discussed last time on the frontend meeting)

File: packages/editor-ui/src/views/WorkflowsView.vue (hunk 1 above)
Comment (tomi) [reply packages/editor-ui/src/views/WorkflowsView.vue:304]: Oops. Will delete

**Review by cstuncsik — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.46.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.46.0)

**Pull request merged**